x-pack/filebeat/module/misp/threat/config/pipeline.js

```diff
@@ -105,71 +114,87 @@ var threat = (function () {
+                    attackPattern = '[' + 'file:hash:sha256 = ' + '\'' + sha256 + '\'' + ' AND file:path = ' + '\'' + filename + '\'' + ']';
+                    attackPatternKQL = 'file.hash.sha256: ' + '"' + sha256 + '"' + ' AND file.path: ' + '"' + filename + '"';
                     evt.Put("file.hash.sha256", sha256);
                     evt.Put("file.path", filename);
                 }
                 break;
             case 'github-username':
-                attackPattern = '[' + 'github.username = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'github:username = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'github.username: ' + '"' + v + '"';
                 break;
             case "hostname":
-                attackPattern = '[' + 'source.domain = ' + '\'' + v + '\'' + ' OR destination.domain = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'source:domain = ' + '\'' + v + '\'' + ' OR destination:domain = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'source.domain: ' + '"' + v + '"' + ' OR destination.domain: ' + '"' + v + '"';
                 evt.Put("source.domain", v);
                 evt.Put("destination.domain", v);
                 break;
             case "ip-dst":
                 ip = v.split("/")[0];
-                attackPattern = '[destination.ip = ' + '\'' + ip + '\'' + ']';
+                attackPattern = '[destination:ip = ' + '\'' + ip + '\'' + ']';
+                attackPatternKQL = 'destination.ip: ' + '"' + ip + '"';
                 evt.Put("destination.ip", ip);
                 break;
             case "ip-dst|port":
                 arr = v.split("|");
                 if (arr.length == 2) {
-                  attackPattern = '[destination.ip = ' + '\'' + arr[0] + '\'' + ' AND destination.port = ' + '\'' + arr[1] + '\'' + ']';
+                  attackPattern = '[destination:ip = ' + '\'' + arr[0] + '\'' + ' AND destination:port = ' + '\'' + arr[1] + '\'' + ']';
+                  attackPatternKQL = 'destination.ip: ' + '"' + arr[0] + '"' + ' AND destination.port: ' + arr[1];
                   evt.Put("destination.ip", arr[0]);
+                  evt.Put("destination.port", arr[1]);
                 }
                 break;
             case "ip-src":
                 ip = v.split("/")[0];
-                attackPattern = '[' + 'source.ip = ' + '\'' + ip + '\'' + ']';
+                attackPattern = '[' + 'source:ip = ' + '\'' + ip + '\'' + ']';
+                attackPatternKQL = 'source.ip: ' + '"' + ip + '"';
                 evt.Put("source.ip", ip);
                 break;
             case "link":
-                attackPattern = '[' + 'url.full = ' + '\'' + v + '\'' + ']';
+                attackPattern = '[' + 'url:full = ' + '\'' + v + '\'' + ']';
+                attackPatternKQL = 'url.full: ' + '"' + v + '"';
```

Comment:
You might also want to search the `http.request.referrer` URL?

Reply:
The http record, that has the referrer URL as the url.full field, should have already matched the query. So this might not be needed.